controllers/rack.go

```diff
@@ -60,6 +59,39 @@ func (r *SingleClusterReconciler) reconcileRacks() reconcileResult {
 		ignorablePodNames,
 	)
 
+	// handle failed racks
+	for idx := range rackStateList {
+		var podList []*corev1.Pod
+
+		state := &rackStateList[idx]
+		found := &appsv1.StatefulSet{}
+		stsName := getNamespacedNameForSTS(r.aeroCluster, state.Rack.ID)
+
+		if err = r.Client.Get(context.TODO(), stsName, found); err != nil {
+			if !errors.IsNotFound(err) {
+				return reconcileError(err)
+			}
+
+			continue
+		}
+
+		podList, err = r.getOrderedRackPodList(state.Rack.ID)
+		if err != nil {
+			return reconcileError(
+				fmt.Errorf(
+					"failed to list pods: %v", err,
+				),
+			)
+		}
+
+		failedPods, _ := getFailedAndActivePods(podList)
+		if len(failedPods) != 0 {
+			if res = r.upgradeOrRollingRestartRack(found, state, ignorablePods, failedPods); !res.isSuccess {
+				return res
+			}
```

Comment:
Can we add some logging here to denote that upgrade/rollingRestart is happening for failed pods?